### ISSUE TITLE
Free vars in qualifiers: LocalHole.hs

### ISSUE THREAD

**Issue by bmcfluff** [ucsd-progsys/liquidhaskell#526]
n$35$a18B is neither an argument nor a global constant in this generated qualifier:

qualif Auto(v$35$16 : int): (v$35$16 <= n$35$a18B) // "tests/pos/LocalHole.hs" (line 10, column 5)

(See #525)

**Comment by nikivazou** [comment]
@gridaphobe can you look at this? It is related with local variables appearing in the type signatures and thus in qualifiers.

**Comment by ranjitjhala** [comment]
I can look at this I'm pretty sure the bugg is in qualifiers.HS

**Comment by nikivazou** [comment]
It is not exactly a bug. The code is 

```
mysum xs = go 0 0
  where
    n = length xs
    {-@ go :: i:{Nat | i <= n} -> _ -> _ @-}
    go i acc
      | i >= n    = acc
      | otherwise = go (i+1) (acc + xs !! i)
```

where `n` is a local variable that appears free in the qualifiers because it is not bound by the type signature.

**Comment by ranjitjhala** [comment]
Yes I realize that. The "bug" is that n should have been made a parameter of the qualif. I think I know why it wasn't...

**Comment by ranjitjhala** [comment]
My guess is that it wasn't because we only generalize the variables  that are bound insight the type specification not those that come from outside (because we don't know their sort)

**Comment by ranjitjhala** [comment]
Fixed by 96bfc6ea705fb00c75cda802d8cb0a330daa7515
